core/trino-spi/src/main/java/io/trino/spi/PageBuilder.java

```diff
@@ -85,6 +86,22 @@ public void reset()
         }
     }
 
+    /**
+     * Rolls back added data to the specified position.
+     * Resetting may result in a block without nulls with the may-have-nulls flag set.
+     * The PageBuilder status will not be updated to reflect the removed data size.
+     *
+     * @throws IllegalArgumentException if the position is greater than the current position count
+     */
+    public void resetTo(int position)
+    {
+        checkIndex(position, declaredPositions);
```

Comment:
I think you need `checkIndex(position, positionCount + 1);` so you can reset to the current position.

Reply:
@dain, are you sure about that? The reason the block builders do this is because the positionCount in block builders are incremented *after* the entry is closed. From the page builder perspective, nearly all use cases declare the position first, before adding in any entries.

Reply:
Oh I see, if you do +1, it's up to the caller on how they want to manage the position.